libs/plugins/stream-client/src/lib/stage-configurator/stage-configurator.component.ts

```diff
@@ -144,16 +144,56 @@ export class StageConfiguratorComponent implements OnInit, OnDestroy {
     );
     this.initActivitySettings(settingPropsToMap, activitySettings);
 
+    this.outputScope = this.getOutputScope(activitySchema);
     const { outputPropsToMap, outputMappings } = this.getOutputMappingsInfo(
       streamMetadata
     );
-    this.initOutputMappings(outputPropsToMap, outputMappings);
+    this.initOutputMappings(outputPropsToMap, this.outputScope, outputMappings);
 
     this.setSelectTab(settingPropsToMap);
 
     this.open();
   }
 
+  private getInputScope(stageId, state, streamMetadata) {
+    const { mainGraph } = state;
+    const scope = [{ ...streamMetadata }];
+    const prevStageSchema = this.getPrevStageSchema(stageId, mainGraph, state);
+    const schemaOutputs = this.getSchemaOutputs(prevStageSchema, 'previousStage');
+    if (!isEmpty(schemaOutputs)) {
+      scope.push(schemaOutputs);
+    }
+    return scope;
+  }
+
+  private getSchemaOutputs(schema, stage): SchemaOutputs {
+    return {
+      type: schema.type,
+      stage,
+      outputs: schema.outputs,
+    };
+  }
+
+  private getOutputScope(activitySchema) {
+    const scope = [];
+    const currentSchemaOutputs = this.getSchemaOutputs(activitySchema, 'currentStage');
+    scope.push(currentSchemaOutputs);
+    return scope;
+  }
+
+  private getPrevStageSchema(stageId, graph, state) {
+    const { mainItems } = state;
+    let prevStageSchema = {};
+    if (stageId !== graph.rootId) {
+      const selectedNode = graph.nodes[stageId].parents || [];
+      const [selectedNodeParent] = selectedNode;
```

Comment:
we can combine the 2 steps as `const [selectedNodeParent] = graph.nodes[stageId].parents`